openSearchLane.go

```diff
@@ -0,0 +1,277 @@
+package lane
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/opensearch-project/opensearch-go/v3"
+	"github.com/opensearch-project/opensearch-go/v3/opensearchapi"
+)
+
+type LogTemplate struct {
+	AppName      string            `json:"appName"`
+	ParentLaneId string            `json:"parentLaneId,omitempty"`
+	JourneyID    string            `json:"journeyId,omitempty"`
+	LaneID       string            `json:"laneId"`
+	LogMessage   string            `json:"logMessage"`
+	Metadata     map[string]string //TODO
+}
+
+type openSearchLane struct {
+	logLane
+	openSearchConnection *openSearchConnection
+	wg                   sync.WaitGroup
```

Comment:
wg doesn't belong here. Its purpose is to sync one single Close() of some instance.

Consider this flow:

1. lane A created
2. lane B derived from A
3. lane C derived from A
4. lane B closed

In step 4, the Close() returns immediately because lane A and C are still open.

But if you're having A, B and C sharing a wg, then step 4 blocks until lane A and C also close. That's incorrect.

Reply:
In this implementation, each wg operates independently without sharing the same reference. As a result, the NewOpenLane, Derive, and DeriveWithContext methods consistently create a new WaitGroup. Additionally, the Close method always maintains the WaitGroup reference specific to the current lane.